4_gap_analysis/src/1c_N_byState.R

```diff
@@ -0,0 +1,205 @@
+# GAP analysis script for the water quality coordination effort
+# This script covers Nutrients, and designtates fractions and
+#      generic names
+
+library(tidyverse)
+library(feather) # install with install.packages('feather')
+library(scipiper) # install with devtools::install_github('USGS-R/scipiper')
+library(XLConnect) # read xlsx files
+
+#***********************
+# batch mode or manually
+runMode<-'batch'
+if (runMode=='batch'){
+  setwd('D:/abock/NAWQA/WQR/nawqa_wqp') # Set locally for running in batch mode for quick processing of all states
+  args<-commandArgs(trailingOnly = T)
+  state<-args[1]
+}else{
+  state<-'Alabama'
+}
+
+# Cons for this script - Ammonia, nitrate, phosphorus, nitrogen, TKN
+Con<-'TKN'
+StartDate<-'1986-01-01'
+EndDate<-'2015-12-31'
+
+# Designate state abbreviation
+state2<-gsub('_',' ',state)
+stateAb<-state.abb[grep(state2, state.name)][1]
+# State abbreviations for states not covered by R state abbreviation database
+if(state == 'Northern_Mariana_Islands') stateAb='MP'
+if(state == 'Guam') stateAb='GU'
+if(state == 'American_Samoa') stateAb='AS'
+if(state == 'District_of_Columbia') stateAb='DC'
+if(state == 'Puerto_Rico') stateAb='PR'
+if(state == 'Virgin_Islands') stateAb='VI'
+
+# If multiple species generalizations are present (i.e. Nitrate)
+# Each list represents WQP characteristic names that fall under the generic Con/ConFull classification
+if (Con=='Ammonia'){
```

Comment:
These could also be a `switch`, but completely up to you if you want to do that!